mail/common/src/main/java/com/fsck/k9/mail/oauth/AndroidAccountOAuth2TokenStore.java

```diff
@@ -0,0 +1,89 @@
+package com.fsck.k9.mail.oauth;
+
+
+import android.accounts.Account;
+import android.accounts.AccountManager;
+import android.accounts.AccountManagerCallback;
+import android.accounts.AccountManagerFuture;
+import android.app.Activity;
+import android.content.Context;
+import android.os.Bundle;
+
+import com.fsck.k9.mail.AuthenticationFailedException;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+;
+
+/**
+ * An interface between the OAuth2 requirements used for authentication and the AccountManager.
+ */
+public class AndroidAccountOAuth2TokenStore implements OAuth2TokenProvider {
+    private static final String GMAIL_AUTH_TOKEN_TYPE = "oauth2:https://mail.google.com/";
+    private static final String GOOGLE_ACCOUNT_TYPE = "com.google";
+
+    private Map<String,String> authTokens = new HashMap<>();
+    private AccountManager accountManager;
+    private Context context;
+
+    public AndroidAccountOAuth2TokenStore(Context applicationContext) {
+        this.accountManager = AccountManager.get(applicationContext);
+        this.context = context;
+    }
+
+    @Override
+    public void authorizeApi(final String emailAddress, final Activity activity, final OAuth2TokenProviderAuthCallback callback) {
+        Account account = new Account(emailAddress, GOOGLE_ACCOUNT_TYPE);
+        accountManager.getAuthToken(account, GMAIL_AUTH_TOKEN_TYPE, null,
+                activity, new AccountManagerCallback<Bundle>() {
+                    @Override
+                    public void run(AccountManagerFuture<Bundle> accountManagerFuture) {
+                        try {
+                            accountManagerFuture.getResult();
```

Comment:
Get result will throw with 3 kinds of exceptions, Philips code threw a detailed exception with info on what went wrong...